dlib/unicode/unicode.h

```diff
@@ -499,21 +605,71 @@ namespace dlib
     {
         using namespace unicode_helpers;
         ustring temp;
-        std::istringstream sin(str);
 
         temp.reserve(str.size());
 
-        int status;
+        int status = 0;
         unichar ch;
-        while ( (status = u8_to_u32(ch,sin)) > 0)
-            temp.push_back(ch);
+
+        for (size_t i = 0; i < str.size();)
+        {
+            status = u8_to_u32(ch, str.begin() + status, str.end());
```

Comment:
Totally make this function just call the new `convert_utf8_to_utf32()` and accumulate characters into `temp` :D

Reply:
Right, obvious… :sweat_smile: